core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -217,11 +217,14 @@ case class TaskCommitDenied(
  * the task crashed the JVM.
  */
 @DeveloperApi
-case class ExecutorLostFailure(execId: String, isNormalExit: Boolean = false)
-  extends TaskFailedReason {
+case class ExecutorLostFailure(execId: String, isNormalExit: Boolean = false, reason:
+Option[String]) extends TaskFailedReason {
   override def toErrorString: String = {
     val exitBehavior = if (isNormalExit) "normally" else "abnormally"
-    s"ExecutorLostFailure (executor ${execId} exited ${exitBehavior})"
+    s"ExecutorLostFailure (executor ${execId} exited ${exitBehavior})" +
+    reason.fold(""){
```

Comment:
Also this line should be indented one level down; I almost missed the `+` on the previous line because of that.